hooks/use-user.ts

```diff
@@ -0,0 +1,33 @@
+"use client";
+import { supabase } from "@/utils/supabase";
+import { useSession } from "next-auth/react";
+import { useCallback, useEffect, useState } from "react";
+import { getSession } from "next-auth/react";
+export const useUser = () => {
+  const session = getSession();
+  const { data: sessionData } = useSession();
+
+  const [user, setUser] = useState<any>(null);
+  const handleFetchUser = useCallback(async () => {
+    if (!sessionData) {
+      return;
+    }
+
+    const { data, error } = await supabase.from("user").select("*");
```

Comment:
Generally speaking I like to use my API routes to get information from the database. Where is this data coming from?